### PR TITLE
grafana/toolkit: improve CI task

### PR DESCRIPTION
**Pull request by ryantxu**
This PR:
* writes to plugin.info.build rather than plugin.info.source
* includes the PR link in the report if it exists
* uploads a report to s3 buckets (using credentials configured in circleci)
* uses `lastIndexOf()` [to find modules](https://github.com/grafana/grafana/pull/18189/commits/73395c01f8e2edbf5055621ba8a0ad79afe2f6fc) -- this helps when building go files within double src folders

### CODE REVIEW

File: packages/grafana-toolkit/src/cli/tasks/plugin.ci.ts [hunk 1]
```diff
@@ -239,8 +248,12 @@ const testPluginRunner: TaskRunner<PluginCIOptions> = async ({ full }) => {
 
     console.log('Grafana: ' + JSON.stringify(results.grafana, null, 2));
 
-    const pluginSettings = await axios.get(`api/plugins/${pluginInfo.id}/settings`, args);
-    console.log('Plugin Info: ' + JSON.stringify(pluginSettings.data, null, 2));
+    const loadedMetaRsp = await axios.get(`api/plugins/${pluginInfo.id}/settings`, args);
+    const loadedMeta: PluginMeta = loadedMetaRsp.data;
+    console.log('Plugin Info: ' + JSON.stringify(loadedMeta, null, 2));
+    if (loadedMeta.info.build) {
+      console.log('Compage to', pluginInfo.info.build);
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/tasks/plugin.ci.ts]: `Compage`?

File: packages/grafana-ui/src/types/plugin.ts [hunk 2]
```diff
@@ -96,7 +101,7 @@ export interface PluginMetaInfo {
     small: string;
   };
   build?: PluginBuildInfo;
-  screenshots: any[];
+  screenshots: ScreenshotInfo[];
```
Comment (ryantxu) [inline packages/grafana-ui/src/types/plugin.ts]: this is the only change that is not in toolkit

**Comment by ryantxu** [comment]
@torkelo -- can you take a quick look?  This is all in toolkit, and I think can safely be merged even though it is still needs work.

**Review by torkelo — APPROVED** [review]

**Comment by torkelo** [comment]
Sorry, late review.

File: packages/grafana-toolkit/src/cli/tasks/toolkit.build.ts [hunk 3]
```diff
@@ -111,8 +115,23 @@ const toolkitBuildTaskRunner: TaskRunner<void> = async () => {
   await preparePackage(pkg);
   fs.mkdirSync('./dist/bin');
   fs.mkdirSync('./dist/sass');
-  await moveFiles();
+  await copyFiles();
   await copySassFiles();
+
+  // RYAN HACK HACK HACK
+  // when Dominik is back from vacation, we can find a better way
```
Comment (dprokop) [inline packages/grafana-toolkit/src/cli/tasks/toolkit.build.ts]: <3

**Pull request merged**